### PR TITLE
Add cilium/external-auth step for hcp

### PR DESCRIPTION
**Pull request by yingzhanredhat**

### CODE REVIEW

**Comment by yingzhanredhat** [comment]
pj-rehearse

**Comment by yingzhanredhat** [comment]
/pj-rehearse pull-ci-openshift-rosa-master-e2e-presubmits-pr-rosa-hcp-advanced

**Comment by yingzhanredhat** [comment]
/pj-rehearse periodic-ci-openshift-rosa-master-e2e-rosa-hcp-external-auth-critical-high-f3

**Comment by yingzhanredhat** [comment]
/pj-rehearse pull-ci-openshift-rosa-master-e2e-presubmits-pr-rosa-sts-advanced

File: ci-operator/step-registry/rosa/setup/conf/cilium/rosa-setup-conf-cilium-commands.sh [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    chainingMode: portmap
+  prometheus:
+    serviceMonitor: {enabled: false}
+  hubble:
+    tls: {enabled: false}
+  sessionAffinity: true
+' | envsubst > /tmp/ciliumconfig.json
+
+oc apply -f /tmp/ciliumconfig.json
+oc wait --for=condition=Ready pod -n cilium --all --timeout=5m
```
Comment (yasun1) [inline ci-operator/step-registry/rosa/setup/conf/cilium/rosa-setup-conf-cilium-commands.sh:114]: Could us use the OCP step https://github.com/openshift/release/blob/master/ci-operator/step-registry/cucushift/hypershift-extended/cilium/cucushift-hypershift-extended-cilium-commands.sh?

File: ci-operator/step-registry/rosa/setup/conf/external-auth/rosa-setup-conf-external-auth-ref.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    documentation: Wait for all worker nodes to be ready. This step is always the final step in the pre part of a workflow.
+  - name: REPLICAS
+    default: "2"
+    documentation: Number of compute nodes to provision. Single zone clusters need at least 2 nodes, multizone clusters need at least 3 nodes. It is only valid while enable_autoscaling is false.
+  - name: ENABLE_AUTOSCALING
+    default: "false"
+    documentation: Set to 'true' to enable autoscaling of compute nodes.
+  - name: MIN_REPLICAS
+    default: "2"
+    documentation: The min number of the compute nodes. It is valid when the enable_autoscaling is true. The value should be a multiple of three for multizone clusters.
```
Comment (yasun1) [inline ci-operator/step-registry/rosa/setup/conf/external-auth/rosa-setup-conf-external-auth-ref.yaml:25]: Please remove the useless variables, e.g. `REPLICAS`

File: ci-operator/step-registry/rosa/setup/conf/cilium/rosa-setup-conf-cilium-commands.sh (hunk 1 above)
Comment (yingzhanredhat) [reply ci-operator/step-registry/rosa/setup/conf/cilium/rosa-setup-conf-cilium-commands.sh]: I think we can't use it beauce we need to check the cluster configures and only do it for hcp cluster with other network type

**Comment by yingzhanredhat** [comment]
/pj-rehearse periodic-ci-openshift-rosa-master-e2e-rosa-hcp-external-auth-critical-high-f3

**Comment by yingzhanredhat** [comment]
/pj-rehearse periodic-ci-openshift-rosa-master-e2e-rosa-hcp-external-auth-critical-high-f3

**Comment by yingzhanredhat** [comment]
/pj-rehearse periodic-ci-openshift-rosa-master-e2e-rosa-hcp-external-auth-critical-high-f3

File: ci-operator/step-registry/rosa/lifecycle/rosa-lifecycle-workflow.yaml [hunk 3]
```diff
@@ -4,6 +4,8 @@ workflow:
     pre:
     - ref: rosa-setup
     - ref: rosa-setup-readiness-cluster
+    - ref: rosa-setup-conf-external-auth
+    - ref: rosa-setup-conf-cilium
```
Comment (yasun1) [inline ci-operator/step-registry/rosa/lifecycle/rosa-lifecycle-workflow.yaml:8]: I do not think adding an extraordinary step to the typical workflow is an excellent way. A better way to do this is to start a new workflow. The workflow is a fantastic way to help us with different configurations; it is complex and unreadable to combine all of the logic into one.

File: ci-operator/step-registry/rosa/setup/conf/cilium/rosa-setup-conf-cilium-commands.sh (hunk 1 above)
Comment (yasun1) [reply ci-operator/step-registry/rosa/setup/conf/cilium/rosa-setup-conf-cilium-commands.sh]: As I said https://github.com/openshift/release/pull/58252#discussion_r1825651837, we could add the step in a new workflow

**Pull request closed without merging**